core/src/main/java/hudson/model/Run.java

```diff
@@ -1541,6 +1541,20 @@ public ACL getACL() {
         return getParent().getACL();
     }
 
+    /**
+     * Deletes this build's entire log, but keeps build
+     *
+     * @throws IOException
+     *            if we fail to delete.
+     */
+    public void deleteLog() throws IOException {
+        File rootDir = getRootDir();
+        LOGGER.log(FINE, "{0}: {1} logs successfully deleted", new Object[] { this, rootDir });
+        synchronized (this) {
+            Util.deleteFile(getLogFile());
```

Comment:
It makes an assumption that the logs are stored on the disk.
Some plugins like Logstash plugin do not actually do that, especially if there is External Logging implementation like https://github.com/jenkinsci/logstash-plugin/pull/18 in place

Not a big problem for this PR as long as the method is overridable. 
But it would great to ensure it does not blow up if the log file does not exist